ambry-server/src/main/java/com/github/ambry/server/RepairRequestsSender.java

```diff
@@ -0,0 +1,242 @@
+
+        boolean hasMore = true;
+        boolean hasError = false;
+        // if some partitions have more requests, continue to handle them.
+        while (hasMore && !hasError) {
+          hasMore = false;
+          hasError = false;
+          // loop all the partitions. For each partition, we handle maxResults number of requests.
+          for (long partitionId : partitionIds) {
+            if (shutdown) {
+              return;
+            }
+            // get the repair requests for this partition
+            List<RequestInfo> requestInfos = getAmbryRequest(partitionId);
+            if (requestInfos.isEmpty()) {
+              continue;
+            }
+            // this partition may have more requests.
+            if (requestInfos.size() >= maxResults) {
+              hasMore = true;
+            }
+
+            Set<Integer> requestsToDrop = new HashSet<>();
+            // send the repair requests to the handler and wait for the responses.
+            for (RequestInfo reqInfo : requestInfos) {
+              List<ResponseInfo> responses =
+                  client.sendAndPoll(Collections.singletonList(reqInfo), requestsToDrop, POLL_TIMEOUT_MS);
+              if (responses == null || responses.size() == 0) {
+                logger.error("RepairRequests Sender: timeout waiting for repairs {} {}", nodeId, reqInfo.getRequest());
+                hasError = true;
+                break;
+              }
+              ResponseInfo resInfo = responses.get(0);
+              ReplicateBlobResponse res = (ReplicateBlobResponse) resInfo.getResponse();
+              ReplicateBlobRequest req = (ReplicateBlobRequest) resInfo.getRequestInfo().getRequest();
+              if (res.getError() == ServerErrorCode.No_Error) {
+                RepairRequestRecord.OperationType type;
+                if (req.getOperationType() == RequestOrResponseType.TtlUpdateRequest) {
+                  type = RepairRequestRecord.OperationType.TtlUpdateRequest;
+                } else {
+                  type = RepairRequestRecord.OperationType.DeleteRequest;
+                }
+                db.removeRepairRequests(req.getBlobId().toString(), type);
+                logger.info("RepairRequests Sender: Repaired {}", req);
+              } else {
+                // LOCAL_CONSISTENCY_TODO: metrics for all the errors happens in RepairRequestsSender.
+                // Ideally we may need create alert if we cannot repair the requests in timely matter.
+                logger.error("RepairRequests Sender: failed to repair {} response {}", req, res);
+                hasError = true;
+              }
+            }
+          } // loop all the partitions
+        }  // loop until all the partitions are clean
+      } catch (Throwable e) {
+        // LOCAL_CONSISTENCY_TODO add metric to track background threads
+        logger.error("RepairRequests Sender: Exception when handling request", e);
+      } finally {
+      }
```

Comment:
we don't need the finally statement here.

Reply:
Will remove it in the metrics PR.